modyn/supervisor/internal/grpc_handler.py

```diff
@@ -200,18 +202,52 @@ def register_pipeline_at_selector(self, pipeline_config: dict) -> int:
         if not self.connected_to_selector:
             raise ConnectionError("Tried to register pipeline at selector, but no connection was made.")
 
+        if "config" in pipeline_config["model"]:
+            model_config = json.dumps(pipeline_config["model"]["config"])
+        else:
+            model_config = "{}"
+
+        model_storage_config = pipeline_config["model_storage"]
+        incremental_model_strategy: Optional[StrategyConfig] = None
+        full_model_interval: Optional[int] = None
+        if "incremental_model_strategy" in model_storage_config:
+            incremental_strategy = model_storage_config["incremental_model_strategy"]
+            incremental_model_strategy = self.get_model_strategy(incremental_strategy)
```

Comment:
couldn't we also extend the constructor of ModelStoragePOlicyINfo to also support instantiation from a dict like this?

Reply:
Maybe I'm overlooking something. Here, we instantiate a grpc request. For this purpose, we read from the pipeline config and add the information to the fields of the request. `ModelStoragePolicyInfo` refers to one message of the proto file.